dace/libraries/blas/nodes/gemm.py

```diff
@@ -425,6 +425,575 @@ def expansion(node, state, sdfg):
+            if node.beta != 0:
+                if node.beta != 1.0:
+                    add_prev_c = f" + {node.beta} * prev_c"
+                else:
+                    add_prev_c = " + prev_c"
+            else:
+                add_prev_c = ""
+            tasklet_inputs = {"from_kernel", "prev_c"
+                              } if node.beta != 0 else {"from_kernel"}
+            tasklet = state.add_tasklet(
+                "write_C", tasklet_inputs, {"to_memory"}, f"""\
+if tm * {T} + m  < {M}  and  n0 * {P} + n1 < {N} :                                               
+    to_memory = {mul_accumulated}{add_prev_c}
+""")
+            state.add_memlet_path(pipe,
+                                  entry_map,
+                                  tasklet,
+                                  dst_conn="from_kernel",
+                                  memlet=dace.Memlet(f"C_pipe[{P}-1]"))
+            if node.beta != 0:
+                state.add_memlet_path(mem_read,
+                                      entry_map,
+                                      tasklet,
+                                      dst_conn="prev_c",
+                                      memlet=dace.Memlet(
+                                          f"_cin[n0 * {P} + n1, tm * {T} + m]",
+                                          dynamic=True,
+                                          allow_oob=True))
+
+            state.add_memlet_path(tasklet,
+                                  exit_map,
+                                  mem,
+                                  src_conn="to_memory",
+                                  memlet=dace.Memlet(
+                                      f"_c[n0 * {P} + n1, tm * {T} + m]",
+                                      dynamic=True,
+                                      allow_oob=True))
+
+        def make_compute(sdfg, state):
+
+            A_pipe_in = state.add_read("A_pipe")
+            B_pipe_in = state.add_read("B_pipe")
+            B_pipe_out = state.add_write("B_pipe")
+            C_pipe_in = state.add_read("C_pipe")
+            C_pipe_out = state.add_write("C_pipe")
+
+            # The computation is expressed a single, flattened loop, which is generated by the following
+            # pipeline scope. Each PE accumulates over T partial results. The drain phase last P*T clock cycles.
+            # Draining and compute are overlapped.
+            # We are generating the loop by explicitly ignoring loop carried dependencies. Therefore, we have
+            # to guarantee that the PE will accumulate on the same partial result only when its value is consolidated.
+            # The + L is a safe delay between accumulation between the same partial result.
+            # It must be computed by considering T and the latency needed to consolidate a partial result
+            # (which is the latency of a multiply-add + latency for reading and writing to BRAM).
```

Comment:
I think it's only [add + BRAM latency], because the multiplication can be precomputed (it does not depend on what's already in the destination register).